source/contracts/reporting/ReputationToken.sol

```diff
@@ -30,34 +30,58 @@ contract ReputationToken is DelegationTarget, ITyped, Initializable, VariableSup
         return true;
     }
 
-    // AUDIT: check for reentrancy issues here, _destination will be called as contracts during validation
+    function migrateOutStakeToken(IReputationToken _destination, address _reporter, uint256 _attotokens) public afterInitialized returns (bool) {
+        require(universe.isContainerForStakeToken(IStakeToken(msg.sender)));
+        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, true);
+    }
+
+    function migrateOutDisputeBondToken(IReputationToken _destination, address _reporter, uint256 _attotokens) public afterInitialized returns (bool) {
+        require(universe.isContainerForDisputeBondToken(IDisputeBond(msg.sender)));
+        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, true);
+    }
+
     function migrateOut(IReputationToken _destination, address _reporter, uint256 _attotokens) public afterInitialized returns (bool) {
+        return internalMigrateOut(_destination, msg.sender, _reporter, _attotokens, false);
+    }
+
+    // AUDIT: check for reentrancy issues here, _destination will be called as contracts during validation
+    function internalMigrateOut(IReputationToken _destination, address _sender, address _reporter, uint256 _attotokens, bool _bonusIfInForkWindow) internal returns (bool) {
```

Comment:
This should probably be `private` rather than `internal`.  In general, I'm of the school of thought that things should only be `internal` if the class/contract is intentionally engineered to be derived from.  ReputationToken is not intended to be derived from.

Reply:
Will change